src/shared/ui/solid-button/SolidButton.tsx

```diff
@@ -0,0 +1,71 @@
+'use client';
+
+import type { ComponentProps, ButtonHTMLAttributes } from 'react';
+import { forwardRef } from 'react';
+import { SurfIcon } from '@/shared/ui/icon/SurfIcon';
+
+type SurfIconName = ComponentProps<typeof SurfIcon>['name'];
+
+type ButtonSize = 's' | 'm' | 'l';
+type ButtonVariant = 'primary' | 'secondary' | 'danger' | 'warning';
+
+type SolidButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
+  size: ButtonSize;
+  variant: ButtonVariant;
+  leftIconName?: SurfIconName | null;
+  rightIconName?: SurfIconName | null;
+  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
+};
+
+export const SolidButton = forwardRef<HTMLButtonElement, SolidButtonProps>(function SolidButton({
+  size = 'm',
+  variant = 'primary',
+  disabled = false,
+  leftIconName,
+  rightIconName,
+  children,
+  type = 'button',
+  onClick,
+  ...rest
+}: SolidButtonProps) {
+  const sizeHeightMap: Record<ButtonSize, string> = {
+    s: 'h-[2rem]',
+    m: 'h-[2.5rem]',
+    l: 'h-[3rem]',
+  };
+
+  const sizeTextMap: Record<ButtonSize, string> = {
+    s: 'text-body-14-600--1-20',
+    m: 'text-body-16-600--1',
+    l: 'text-body-16-600--1',
+  };
+
+  const variantMap: Record<ButtonVariant, string> = {
+    primary: 'text-foreground-accent bg-background-primary hover:bg-foreground-primary',
+    secondary: 'text-foreground-normal bg-background-tertiary hover:bg-background-quaternary',
+    danger: 'text-foreground-accent bg-foreground-danger hover:bg-foreground-danger-darker',
+    warning: 'text-foreground-danger bg-background-tertiary hover:bg-background-quaternary',
+  };
+
+  const disabledClass =
+    'bg-background-tertiary text-foreground-accent cursor-not-allowed opacity-50';
+
+  return (
+    <button
+      type={type}
+      disabled={disabled}
+      onClick={onClick}
+      {...rest}
```

Comment:
@rngrhn4114 forwardRef를 사용시 props와 ref를 함께 넘겨줘야 한다고 합니다! 확인 부탁드립니다.
```suggestion
      {...rest}
      ref={ref}
```

Reply:
넵 반영한 버전이 지금 alert-component 브랜치에 있어서 일단 로컬에서는 임시로 반영해서 테스트 해보시면 될 것 같숩니다

Reply:
아하 반영되었군요~ 감사합니다!!